src/main/java/seedu/address/logic/ai/Ai.java

```diff
@@ -143,6 +144,24 @@ public static double getDistanceMetric(HashMap<String, Double> featureMap1, Hash
         return sort(distance);
     }
 
+    /**
+     * Returns the common languages for each user
+     *
+     * @param mainUserStat the user to compare with
+     * @param stats stats of all the users
+     * @return a dictionary of users and common languages
+     */
+    public static HashMap<String, ArrayList<String>> getCommonLanguages(
+            HashMap<String, Double> mainUserStat, HashMap<String, HashMap<String, Double>> stats) {
+        HashMap<String, ArrayList<String>> commonLanguages = new HashMap<>();
+        stats.keySet().forEach(u -> commonLanguages.put(u, new ArrayList<>(sort(stats.get(u))
+                .keySet().stream()
+                .filter(s -> !s.equals("repo-count") && !s.equals("Other") && mainUserStat.containsKey(s))
```

Comment:
Multiple usages of string `"repo-count"`, a noticeable number of times in method `normalizeRepoCount`. 
I think you should set `"repo-count"` as a boolean string variable.